python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -249,53 +287,75 @@ def predict_proba(self, dataset: pd.DataFrame) -> np.ndarray:
+    def predict(self, x, iou_thresh: float = 0.5, score_thresh: float = 0.5):
+        """Create a list of detection records from the image predictions.
 
-        :param dataset: The dataset to predict on.
-        type dataset: PIL Image or numpy.ndarray
-        param device: device pytorch is writing to
-        type device: pytorch device
-        param iou_threshold: amount of acceptable error.
-            objects with error scores higher than the threshold will be removed
-        type iou_threshold: float
-        """
-        predictions = []
-        for image in dataset:
-            detections = self._model(ToTensor()(image).to(self._device).unsqueeze(0))
+        :param x: Tensor of the image
+        :type x: torch.Tensor
+        :return: Baseline detections to get saliency maps for
+        :rtype: List of Detection Records
 
-            prediction = detections[0]
+        Example Label (y) representation for a cohort of 2 images:
 
-            keep = torchvision.ops.nms(
-                prediction["boxes"],
-                prediction["scores"],
-                iou_threshold,
-            )
+        [
 
-            prediction["boxes"] = prediction["boxes"][keep]
-            prediction["scores"] = prediction["scores"][keep]
-            prediction["labels"] = prediction["labels"][keep]
+            [
+            [object_1, x1, y1, b1, h1, (optional) confidence_score],
+            [object_2, x2, y2, b2, h2, (optional) confidence_score],
+            [object_1, x3, y3, b3, h3, (optional) confidence_score]
+        ],
 
-            image_predictions = torch.cat((prediction["labels"].unsqueeze(1),
-                                           prediction["boxes"],
-                                           prediction["scores"].unsqueeze(1)), dim=1)
+        [
+            [object_1, x4, y4, b4, h4, (optional) confidence_score],
+            [object_2, x5, y5, b5, h5, (optional) confidence_score]
+        ]
 
-            predictions.append(image_predictions.detach().cpu().numpy().tolist())
+        ]
 
-        return predictions
+        """
+        detections = []
+        for image in x:
+            if type(image) == torch.Tensor:
+                raw_detections = self._model(
+                    image.to(self._device).unsqueeze(0))
+            else:
+                raw_detections = self._model(
+                    T.ToTensor()(image).to(self._device).unsqueeze(0))
+
+            for raw_detection in raw_detections:
+                raw_detection = _apply_nms(raw_detection)
+
+                # Note that FasterRCNN doesn't return a score for each class,
+                # only the predicted class. DRISE requires a score for each.
+                # We approximate the score for each class
+                # by dividing (class score) evenly among the other classes.
+
+                raw_detection = _filter_score(raw_detection)
+                # TODO - add in expanded class scores
```

Comment:
1. What's the utility of this?
2. Adding TODOs is fine, curious if we should have code in comments or remove it as per @imatiach-msft's previous suggestion that we should very rarely have commented code if logic, documentation etc may change and if that code snippet could be rendered incorrect.